src/components/_common/Dropdown/index.tsx

```diff
@@ -1,41 +1,48 @@
 import React from 'react';
 import cn from 'classnames';
 
+import DropdownOption from 'components/_common/DropdownOption';
+import useClickOutside from 'hooks/useClickOutside';
 import styles from './styles.module.scss';
 
-const options = ['release date', 'rating', 'best 100'];
-
 type DropdownProps = {
+  options: string[];
+  isOpen: boolean;
+  value?: string;
   className?: string;
+  onToggle: () => void;
+  onClose: () => void;
+  onOptionClick: (value: string) => void;
 };
 
-const Dropdown: React.FC<DropdownProps> = ({ className }) => {
-  const [value, setValue] = React.useState(options[0]);
-  const [isOpen, setIsOpen] = React.useState(false);
+const Dropdown: React.FC<DropdownProps> = ({
+  options,
+  isOpen,
+  value,
+  className,
+  onOptionClick,
+  onToggle,
+  onClose,
+}) => {
+  const dropdownRef = React.useRef(null);
 
-  const handleOpenDropdownClick = () => {
-    setIsOpen((prevState) => !prevState);
-  };
+  useClickOutside({ ref: dropdownRef, callback: onClose });
 
   return (
-    <div className={cn(styles.root, className)}>
-      <span className={styles.value} onClick={handleOpenDropdownClick}>
-        {value}
-      </span>
-
+    <div className={cn(styles.root, className)} ref={dropdownRef}>
+      {value ? (
```

Comment:
I think we can simplify this code

` <span className={ cn ( value ? styles.value : styles.threeDots) } onClick={onToggle}>
          {value || '...'}
  </span>
`

Reply:
make sense, will fix in the next task